src/wallet/components/Wallet.tsx

```diff
@@ -1,23 +1,22 @@
-import { Children, useMemo, useRef } from 'react';
-import { useIsMounted } from '../../core-react/internal/hooks/useIsMounted';
-import { useTheme } from '../../core-react/internal/hooks/useTheme';
-import { findComponent } from '../../core-react/internal/utils/findComponent';
-import { cn } from '../../styles/theme';
-import { useOutsideClick } from '../../ui/react/internal/hooks/useOutsideClick';
+import { useIsMounted } from '@/core-react/internal/hooks/useIsMounted';
+import { useTheme } from '@/core-react/internal/hooks/useTheme';
+import { findComponent } from '@/core-react/internal/utils/findComponent';
+import { cn } from '@/styles/theme';
+import { useOutsideClick } from '@/ui-react/internal/hooks/useOutsideClick';
+import { Children, useMemo } from 'react';
 import type { WalletReact } from '../types';
 import { ConnectWallet } from './ConnectWallet';
 import { WalletDropdown } from './WalletDropdown';
 import { WalletProvider, useWalletContext } from './WalletProvider';
 
-const WalletContent = ({ children, className }: WalletReact) => {
-  const { isOpen, setIsOpen } = useWalletContext();
-  const walletContainerRef = useRef<HTMLDivElement>(null);
+function WalletContent({ children, className }: WalletReact) {
+  const {
+    isOpen,
+    handleClose,
+    containerRef: walletContainerRef,
```

Comment:
why did you need to move this ref up to the provider?

Reply:
i actually don't anymore, i'll move this back down to the component